engine/crates/engine-v2/schema/src/builder/input_values.rs

```diff
@@ -8,11 +8,14 @@ impl SchemaInputValues {
     pub(crate) fn ingest_arbitrary_federated_value(&mut self, ctx: &BuildContext, value: Value) -> SchemaInputValue {
         match value {
             Value::Null => SchemaInputValue::Null,
-            Value::String(id) => SchemaInputValue::String(id.into()),
+            Value::String(id) | Value::UnboundEnumValue(id) => SchemaInputValue::String(id.into()),
             Value::Int(n) => SchemaInputValue::BigInt(n),
             Value::Float(f) => SchemaInputValue::Float(f),
             Value::Boolean(b) => SchemaInputValue::Boolean(b),
-            Value::EnumValue(id) => SchemaInputValue::String(id.into()),
+            Value::EnumValue(id) => {
+                let id = ctx.idmaps.enum_values.get(id).expect("enum ids to be valid");
```

Comment:
Hmm true, when writing that I was thinking that's basically an invariant and it shouldn't happen, but now rethinking about it, I wouldn't bet that we validate that inaccessible values aren't used in defaults etc.

Reply:
Done. It won't be a hard error in some cases (we just leave the value out) because that is indeed a bigger change, and I think it's fine at that level. If we want to be strict there, that should be validated in composition.

Reply:
>that should be validated in composition.

That's what I used to think until people just used the gateway with their own federated SDL coming from who knows where 😅

Reply:
Good point, yeah, composition _and_ from_sdl, but the schema builder should be able to assume the schema is valid.